Tests/BachandNetworkingTests/HTTPResponseTests.swift

```diff
@@ -95,19 +104,114 @@ final class HTTPResponseFactoryTests: XCTestCase {
+        XCTAssertEqual(nsError.code, HTTPResponseError.Code.schemeNotHTTPS.rawValue)
+      }
+      XCTAssertThrowsError(
+        try makeSecureHTTPResponsePublisher(for: URLRequest(url: url)),
+        "Error has code for .schemeNotHTTPS",
+        errorHandler)
+    }
+  }
+
+  func test_makeSecureHTTPResponsePublisherForURLRequest_urlIsNil_throwsError() throws {
+    try safelyUnwrapURL(string: "url/that/will/be/removed") { url in
+      var urlRequest = URLRequest(url: url)
+      urlRequest.url = nil
+      let errorHandler: (Error) -> Void = {
+        let nsError = $0 as NSError
+        XCTAssertEqual(nsError.code, HTTPResponseError.Code.requestHasNoURL.rawValue)
+      }
+      XCTAssertThrowsError(
+        try makeSecureHTTPResponsePublisher(for: urlRequest),
+        "Error has code for .requestHasNoURL",
+        errorHandler)
+    }
   }
-}
 
-private func makeStubHTTPURLResponse(
-  url: URL = URL(string: "/resource")!,
-  statusCode: Int)
-  -> HTTPURLResponse
-{
-  HTTPURLResponse(
-    url: url,
-    statusCode: statusCode,
-    httpVersion: nil,
-    headerFields: nil)!
+  func test_makeHTTPResponsePublisherForURLRequest_schemeIsHTTP_doesNotThrowsError() throws {
+    try safelyUnwrapURL(string: "http://apple.com") { url in
+      XCTAssertNoThrow(try makeHTTPResponsePublisher(for: URLRequest(url: url)))
+    }
+  }
+
+  func test_makeHTTPResponsePublisherForURLRequest_schemeIsHTTPS_doesNotThrowsError() throws {
+    try safelyUnwrapURL(string: "https://apple.com") { url in
+      XCTAssertNoThrow(try makeHTTPResponsePublisher(for: URLRequest(url: url)))
+    }
+  }
+
+  func test_makeHTTPResponsePublisherForURLRequest_schemeIsFTP_throwsError() throws {
+    try safelyUnwrapURL(string: "ftp://apple.com") { url in
+      let errorHandler: (Error) -> Void = {
+        let nsError = $0 as NSError
+        XCTAssertEqual(nsError.code, HTTPResponseError.Code.schemeNotHTTPBased.rawValue)
+      }
+      XCTAssertThrowsError(
+        try makeHTTPResponsePublisher(for: URLRequest(url: url)),
+        "Error has code for .schemeNotHTTPBased",
+        errorHandler)
+    }
+  }
+
+  func test_makeHTTPResponsePublisherForURLRequest_urlIsNil_throwsError() throws {
+    try safelyUnwrapURL(string: "url/that/will/be/removed") { url in
+      var urlRequest = URLRequest(url: url)
+      urlRequest.url = nil
+      let errorHandler: (Error) -> Void = {
+        let nsError = $0 as NSError
+        XCTAssertEqual(nsError.code, HTTPResponseError.Code.requestHasNoURL.rawValue)
+      }
+      XCTAssertThrowsError(
+        try makeHTTPResponsePublisher(for: urlRequest),
+        "Error has code for .requestHasNoURL",
+        errorHandler)
+    }
+  }
+
+  // Attempts to unwrap a string representing a URL, invoking the provided closure if the unwrapping
+  // is succesful. Fires an XCTest assertion on the calling line of code if the string does not
+  // represent a valid URL.
+  private func safelyUnwrapURL(
```

Comment:
This is what it looks like when the string cannot be unwrapped into a URL.
![Screen Shot 2021-01-10 at 11 41 14 AM](https://user-images.githubusercontent.com/1791049/104134073-568a9d00-533c-11eb-8e35-89e8ae975f09.png)